### ISSUE TITLE
error: Argument 1 to "len" has incompatible type "Optional[List[Any]]"; expected "Sized"

### ISSUE THREAD

**Issue by kev26** [python/mypy#13859]
I defined notation for my function `typing.Union[dict, list, None]`

And in my test case, I passed this test but when I run `mypy`, this raise the error `error: Argument 1 to "len" has incompatible type "Optional[List[Any]]"; expected "Sized"`

my test:
```python
        recipients = get_cronofy_recipient(meeting=meeting)
        self.assertIsNotNone(recipients)
        self.assertIsInstance(recipients, list)
        self.assertEqual(len(recipients),11)
```

**Comment by denballakh** [comment]
It is not a bug.
These lines are now narrowing type of your variable, because mypy cant understand it:
```py
self.assertIsNotNone(recipients)
self.assertIsInstance(recipients, list)
```
So mypy raises `incompatible type` error when you are calling `len(list[Any] | None)`.
If you want narrow type of your var to `list[Any]`, you should write something like this:
```py
self.assertIsNotNone(recipients)
assert recipients is not None # maybe you should put this line into `if TYPE_CHECKING` guarg
self.assertIsInstance(recipients, list)
assert isinstance(recipients, list)
self.assertEqual(len(recipients),11)
```
or (with pytest):
```py
assert recipients is not None
assert isinstance(recipients, list)
```

**Comment by kev26** [comment]
thank you so much !
